didcomm/did_doc/did_resolver.py

```diff
@@ -0,0 +1,56 @@
+from abc import ABC, abstractmethod
+from typing import List, Optional
+
+from didcomm.did_doc.did_doc import DIDDoc
+
+
+class DIDResolver(ABC):
+    """DID Resolver to resolver a DID to a DID DOC."""
+
+    @abstractmethod
+    async def resolve(self, did: str) -> Optional[DIDDoc]:
```

Comment:
What do you think about adding supported did methods to the interface of resolver?

Reply:
What is the purpose of adding it? Who is epxected to know this information or call such a method?

Reply:
It will allow chain resolver to avoid unnecessary calls to resolvers that don't support these methods. Keeping in mind resolve is async function that requires event loop to be called it may save a lot of time for production use cases.